lib/smart_app_launch/token_introspection_response_group.rb

```diff
@@ -0,0 +1,127 @@
+      This group of tests validates the contents of the token introspection response by comparing the fields and/or 
+      values in the token introspection response to the fields and/or values of the original access token response 
+      in which the access token was given to the client.   
+      )
+
+      input_instructions %(
+        There are two categories of input for this test group: 
+
+        1. The access token response values, which will dictate what the tests will expect to find in the token 
+        introspection response.  If the Token Introspection Request test group was run AND the option to introspect
+        the access token from the Standalone Launch tests was selected, these values will auto-fill; otherwise,
+        the tester will need to run an out-of-band ACCESS TOKEN request and manually input the access token response
+        parameters.   
+        
+        2. The token introspection response bodies. If the Token Introspection Request test group was run, these will
+        auto-fill; otherwise, the tester will need to an run out-of-band INTROSPECTION requests for:
+
+          a. The ACTIVE access token received from the out-of-band access token request, AND
+
+          b. An INACTIVE OR INVALID token 
+
+        The client making both introspection requests must be authorized to access the introspection endpoint if
+        the endpoint is protected.
+      )
+      
+    test do 
+      title 'Token introspection response for an active token contains required fields'
+
+      description %(
+        This test will check whether the metadata in the token introspection response is correct for an active token and
+         that the response data matches the data in the original access token and/or access token response from the 
+         authorization server, including the following:
+      
+        Required:
+        *  `active` claim is set to true 
+        * `scope`, `client_id`, and `exp` claim(s) match between introspection response and access token
+
+        Conditionally Required:
+        * IF launch context parameter(s) included in access token, introspection response includes claim(s) for 
+        launch context parameter(s) 
+          * Parameters checked for are `patient` and `encounter`
+        * IF identity token was included as part of access token response, `iss` and `sub` claims are present in 
+        introspection response
+
+        Optional but Recommended:
+        * IF identity token was included as part of access token response, `fhirUser` claim SHOULD be present in 
+        introspection response
+      )
+
+      # Access token value is repeatedly overwritten during refresh token tests and value 
+      # when provided for introspection is from the last refresh token exchange test from standalone launch
+      # But other outputs (id token, received scopes, etc) do not update and are still the same as
+      # they were in the original access token request 
```

Comment:
Hmmm, interesting.  We need to make sure we are careful about this.  I wonder if this test kit should have the token refresh step be a separate thing, like how token introspection is a separate thing.  I'm not sure.

Reply:
The more I think about this, the more I feel that reusing the access token from the standalone launch tests isn't feasible like we originally thought, given the complications with the inputs and the presence of the refresh tokens.  

Instead, I feel it may make more sense to reuse the inputs and/or test methods from the standalone launch test to request a NEW access token as part of the token introspection test setup (perhaps another optional precursor test group that runs before the Issue Introspection Request group?).  This is kind of like what I had originally but it would repeat specific steps of the Standalone Launch test and (hopefully) shouldn't require any new user input.

I know this isn't ideal in a lot of ways, but even if we decided to move the refresh token tests to their own group we would have the same problem of needing an active access token as a basis for the tests; that token would become inactive once it was refreshed, and they don't have long lifespans to begin with.  To put it another way, I worry that a lot of reuse of access tokens would muddle their expected state and potentially complicate tests, whereas starting with a fresh state for each group would remediate that.  I will look into whether this could be done.

Reply:
I added an update to reuse the standalone launch tests in a precursor group for the introspection tests.  Order would be:
1.  Run Test Group 1, Standalone Launch - will auto-populate some fields for 3.1
2. Run Test Group 3.1, Receive Access Token to Introspect - reruns Test Group 1.2 SMART Standalone launch for fresh access token, which will auto-populate fields for 3.2 and 3.3 (access token and expected values for introspection response)
3. Run Test Group 3.2, Issue Introspection Request
4. Run Test Group 3.3, Validate Introspection Response

The upside is it populates all of the information nicely, the big downside is that it makes the input interface for all of Test Group 3 Token Introspection way more confusing.